pkg/stanza/fileconsumer/file.go

```diff
@@ -166,6 +173,17 @@ func (m *Manager) poll(ctx context.Context) {
 	m.rotateFilesets()
 }
 
+func (m *Manager) reportMetrics(ctx context.Context) {
+	var currentlyOpenedFiles int64
+	if m.previousOpenedFiles == 0 {
+		currentlyOpenedFiles = int64(m.totalReaders())
+	} else {
+		currentlyOpenedFiles = int64(m.totalReaders()) - m.previousOpenedFiles
+	}
+	m.previousOpenedFiles = int64(m.totalReaders())
+	m.openFiles.Add(ctx, currentlyOpenedFiles)
```

Comment:
Double check me but can we just do this?
```suggestion
	m.openFiles.Add(ctx, int64(m.totalReaders()) - m.previousOpenedFiles)
	m.previousOpenedFiles = int64(m.totalReaders())
```

Based on two ideas:
1. The if/else isn't necessary because `total - previous` works for either case.
2. We can call `openFiles.Add` before updating `previous`, so don't need the `current` var.